libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+                }
+
+                if (((JObject)sentiment).TryGetValue("score", out var score))
+                {
+                    properties.Add(LuisTelemetryConstants.SentimentScoreProperty, score.Value<string>());
+                }
+            }
+
+            var entities = recognizerResult.Entities?.ToString();
+            properties.Add(LuisTelemetryConstants.EntitiesProperty, entities);
+
+            // Use the LogPersonalInformation flag to toggle logging PII data, text is a common example
+            if (LogPersonalInformation && !string.IsNullOrEmpty(turnContext.Activity.Text))
+            {
+                properties.Add(LuisTelemetryConstants.QuestionProperty, turnContext.Activity.Text);
+            }
+
+            // Additional Properties can override "stock" properties.
+            if (telemetryProperties != null)
+            {
+                return Task.FromResult(telemetryProperties.Concat(properties)
+                           .GroupBy(kv => kv.Key)
+                           .ToDictionary(g => g.Key, g => g.First().Value));
+            }
+
+            return Task.FromResult(properties);
+        }
+
+        private string AddParam(string query, string prop, bool? val)
+        {
+            var result = query;
+            if (val.HasValue)
+            {
+                if (query == null)
+                {
+                    result = $"{prop}={val.Value}";
+                }
+                else
+                {
+                    result += $"&{prop}={val.Value}";
+                }
+            }
+
+            return result;
+        }
+
+        private async Task<RecognizerResult> RecognizeInternalAsync(ITurnContext turnContext, LuisV3PredictionOptions predictionOptions, Dictionary<string, string> telemetryProperties, Dictionary<string, double> telemetryMetrics, CancellationToken cancellationToken)
+        {
+            var options = predictionOptions == null ? _options : MergeDefaultOptionsWithProvidedOptions(_options, predictionOptions);
+
+            BotAssert.ContextNotNull(turnContext);
+
+            if (turnContext.Activity.Type != ActivityTypes.Message)
+            {
+                return null;
+            }
+
+            var utterance = turnContext.Activity?.AsMessageActivity()?.Text;
+            RecognizerResult recognizerResult;
+            JObject luisResponse = null;
+
+            if (string.IsNullOrWhiteSpace(utterance))
+            {
+                recognizerResult = new RecognizerResult
+                {
+                    Text = utterance,
+                    Intents = new Dictionary<string, IntentScore>() { { string.Empty, new IntentScore() { Score = 1.0 } } },
+                    Entities = new JObject(),
+                };
+            }
+            else
+            {
+                var uri = new UriBuilder(_application.Endpoint);
+                uri.Path += $"luis/v3.0-preview/apps/{_application.ApplicationId}";
+
```

Comment:
This seems fragile, as it forces us to rev the SDK when v3.0 comes out. It will also cause this to BREAK once the -preview path goes away.

Reply:
What alternative do you have?  This is a preview feature using a preview REST API whose name will change.

Reply:
I'm leaving this as is with a comment.  When 3.1 happens, other changes will be needed in the code as well.